### DISCUSSION TITLE
Why generated image name so long? Can I specify the name for the generated image?

### DISCUSSION THREAD

**Discussion by epic1219** [devcontainers/spec#464] [General]
The generated image's name is too long, nearly take half of my screen. And when I run `docker ps -a`, the table is a mess cause one container's information takes two rows.

**Comment by samruddhikhandale** [comment]
Hi 👋 

Are you creating the container with the dev container CLI?  If so, have you tried using the `--image-name` option to explicitly specify the image name?

**Reply by epic1219** [reply]
Hi, thanks for the advice.
Actually I didn't use dev container CLI directly, it is managed by vscode. I just wrote `.devcontainer/devcontainer.json` with my customized image.

**Reply by joshspicer** [reply]
FWIW, I find that simply lowering the font a point or two makes the `docker ps` output a lot easier to read.

You may also be interested in [the docker CLI's formatting](https://docs.docker.com/config/formatting/) docs, that could let you visualize your running containers more concisely by providing your own formatting string (you could totally at a shell alias to make this easier to type!)

For example:
```
[~]$ docker ps -a --format '{{.ID}} {{.Names}}'
9c32c5a54dd4 serene_sammet
0cab811e0910 compassionate_hawking
074572a675fa buildx_buildkit_remote0
671ae88f75a3 vigorous_wing
a1440cda9028 charming_maxwell
0938e87f3786 frosty_lumiere
2d82f65ff6ad happy_shockley
af33c67beebb buildkitd
```

**Reply by epic1219** [reply]
Yeah, that will do. Since you dropped the image name part. But I think image and container name should always be displayed.

>
    [~]$ docker images -a --format 'table {{.Repository}}\t{{.Tag}}'
    REPOSITORY                                                                      TAG
    vsc-proj-3e60c91bdc1d909edd1fc361d8ecb0d31dd68062d6584622f55aa2423332e776-uid   latest
    infra/db/mysql                                                                  8.0.36
    debian                                                                          bookworm

    [~]$ docker ps -a --format 'table {{.ID}}\t{{.Names}}\t{{.Image}}'
    CONTAINER ID   NAMES               IMAGE
    f5a0b785bc56   mysql8              infra/db/mysql:8.0.36
    fb83dbb6dd63   charming_shockley   vsc-proj-3e60c91bdc1d909edd1fc361d8ecb0d31dd68062d6584622f55aa2423332e776-uid:latest

**Reply by Mortiemi** [reply]
> Yeah, that will do. Since you dropped the image name part. But I think image and container name should always be displayed.
> 
> 
> 
> >
> 
>     [~]$ docker images -a --format 'table {{.Repository}}\t{{.Tag}}'
> 
>     REPOSITORY                                                                      TAG
> 
>     vsc-proj-3e60c91bdc1d909edd1fc361d8ecb0d31dd68062d6584622f55aa2423332e776-uid   latest
> 
>     infra/db/mysql                                                                  8.0.36
> 
>     debian                                                                          bookworm
> 
> 
> 
>     [~]$ docker ps -a --format 'table {{.ID}}\t{{.Names}}\t{{.Image}}'
> 
>     CONTAINER ID   NAMES               IMAGE
> 
>     f5a0b785bc56   mysql8              infra/db/mysql:8.0.36
> 
>     fb83dbb6dd63   charming_shockley   vsc-proj-3e60c91bdc1d909edd1fc361d8ecb0d31dd68062d6584622f55aa2423332e776-uid:latest
> 
>

**Reply by Mortiemi** [reply]
> > Yeah, that will do. Since you dropped the image name part. But I think image and container name should always be displayed.
> > 
> > 
> > 
> > >
> > 
> >     [~]$ docker images -a --format 'table {{.Repository}}\t{{.Tag}}'
> > 
> >     REPOSITORY                                                                      TAG
> > 
> >     vsc-proj-3e60c91bdc1d909edd1fc361d8ecb0d31dd68062d6584622f55aa2423332e776-uid   latest
> > 
> >     infra/db/mysql                                                                  8.0.36
> > 
> >     debian                                                                          bookworm
> > 
> > 
> > 
> >     [~]$ docker ps -a --format 'table {{.ID}}\t{{.Names}}\t{{.Image}}'
> > 
> >     CONTAINER ID   NAMES               IMAGE
> > 
> >     f5a0b785bc56   mysql8              infra/db/mysql:8.0.36
> > 
> >     fb83dbb6dd63   charming_shockley   vsc-proj-3e60c91bdc1d909edd1fc361d8ecb0d31dd68062d6584622f55aa2423332e776-uid:latest
> > 
> > 
> 
>

**Reply by zuriel5zurielg20** [reply]
> Yeah, that will do. Since you dropped the image name part. But I think image and container name should always be displayed.
> 
> > 
> 
> ```
> [~]$ docker images -a --format 'table {{.Repository}}\t{{.Tag}}'
> REPOSITORY                                                                      TAG
> vsc-proj-3e60c91bdc1d909edd1fc361d8ecb0d31dd68062d6584622f55aa2423332e776-uid   latest
> infra/db/mysql                                                                  8.0.36
> debian                                                                          bookworm
> 
> [~]$ docker ps -a --format 'table {{.ID}}\t{{.Names}}\t{{.Image}}'
> CONTAINER ID   NAMES               IMAGE
> f5a0b785bc56   mysql8              infra/db/mysql:8.0.36
> fb83dbb6dd63   charming_shockley   vsc-proj-3e60c91bdc1d909edd1fc361d8ecb0d31dd68062d6584622f55aa2423332e776-uid:latest
> ```
,)

**Reply by Mortiemi** [reply]
> Yeah, that will do. Since you dropped the image name part. But I think image and container name should always be displayed.
> 
> 
> 
> >
> 
>     [~]$ docker images -a --format 'table {{.Repository}}\t{{.Tag}}'
> 
>     REPOSITORY                                                                      TAG
> 
>     vsc-proj-3e60c91bdc1d909edd1fc361d8ecb0d31dd68062d6584622f55aa2423332e776-uid   latest
> 
>     infra/db/mysql                                                                  8.0.36
> 
>     debian                                                                          bookworm
> 
> 
> 
>     [~]$ docker ps -a --format 'table {{.ID}}\t{{.Names}}\t{{.Image}}'
> 
>     CONTAINER ID   NAMES               IMAGE
> 
>     f5a0b785bc56   mysql8              infra/db/mysql:8.0.36
> 
>     fb83dbb6dd63   charming_shockley   vsc-proj-3e60c91bdc1d909edd1fc361d8ecb0d31dd68062d6584622f55aa2423332e776-uid:latest
> 
>
